### PR TITLE
Test: Merge upstream ConditionalFact changes

### PR DESCRIPTION
**Pull request by bricelam**
[appveyor test]

### CODE REVIEW

File: src/EFCore.Specification.Tests/TestUtilities/Xunit/FrameworkSkipConditionAttribute.cs [hunk 1]
```diff
@@ -27,20 +26,17 @@ private static bool CanRunOnThisFramework(RuntimeFrameworks excludedFrameworks)
...
+#if NET452
+            if (excludedFrameworks.HasFlag(RuntimeFrameworks.CLR))
             {
                 return false;
             }
-
-            if (excludedFrameworks.HasFlag(RuntimeFrameworks.CoreCLR)
-                &&
-                RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase))
+#else
```
Comment (smitpatel) [inline src/EFCore.Specification.Tests/TestUtilities/Xunit/FrameworkSkipConditionAttribute.cs]: How about conditional If? combines both return false.

Comment (bricelam) [inline src/EFCore.Specification.Tests/TestUtilities/Xunit/FrameworkSkipConditionAttribute.cs]: I like it! But... I'll keep it the same as the upstream for now.

**Comment by bricelam** [comment]
Upstream code is in [aspnet/Testing](https://github.com/aspnet/Testing) although we've deviated a bit.

**Comment by smitpatel** [comment]
nit: our ConditionalAttribute can also use nameof pattern like upstream.

**Pull request merged**